.github/workflows/composite/cartopy-cache/action.yml

```diff
@@ -0,0 +1,41 @@
+name: "cartopy-cache"
+description: "create and cache cartopy assets"
+
+#
+# Assumes the environment contains the following variables:
+#   - CONDA
+#
+inputs:
+  cache_build:
+    description: "conda environment cache build number"
+    required: false
+    default: "0"
+  cache_period:
+    description: "conda environment cache timestamp"
+    required: true
+  env_name:
+    description: "environment name"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/cache@v3
+      id: cartopy-cache
+      with:
+        path: ~/.local/share/cartopy
+        key: ${{ runner.os }}-cartopy-${{ inputs.env_name }}-p${{ inputs.cache_period }}-b${{ inputs.cache_build }}
+
+    - if: steps.cartopy-cache.outputs.cache-hit != 'true'
+      env:
+        CARTOPY_SHARE_DIR: ~/.local/share/cartopy
+        CARTOPY_FEATURE: https://raw.githubusercontent.com/SciTools/cartopy/v0.20.0/tools/cartopy_feature_download.py
+      shell: bash
+      run: |
+        # Require to explicitly activate the environment within the composite action.
+        source ${{ env.CONDA }}/etc/profile.d/conda.sh >/dev/null 2>&1
+        conda activate ${{ inputs.env_name }}
+        wget --quiet ${CARTOPY_FEATURE}
+        mkdir -p ${CARTOPY_SHARE_DIR}
+        # Requires a pre-installed version of cartopy within the environment.
+        python cartopy_feature_download.py physical --output ${CARTOPY_SHARE_DIR} --no-warn
```

Comment:
@ESadek-MO This has changed for the better with `cartopy` v0.23+, see https://github.com/SciTools/cartopy/pull/2263

However, we can change this at a later date. The current implementation will still always work 👍